cmd/kubeadm/app/master/selfhosted.go

```diff
@@ -340,3 +399,277 @@ func getSchedulerDeployment(cfg *kubeadmapi.MasterConfiguration, volumes []v1.Vo
 func buildStaticManifestFilepath(name string) string {
 	return path.Join(kubeadmapi.GlobalEnvParams.KubernetesDir, "manifests", name+".yaml")
 }
+
+func apiServerProjectedVolume() v1.Volume {
+	return v1.Volume{
+		Name: "k8s",
+		VolumeSource: v1.VolumeSource{
+			Projected: &v1.ProjectedVolumeSource{
+				Sources: []v1.VolumeProjection{
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.CACertAndKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  "tls.crt",
+									Path: path.Join("pki", kubeadmconstants.CACertName),
+								},
+								{
+									Key:  "tls.key",
+									Path: path.Join("pki", kubeadmconstants.CAKeyName),
+								},
+							},
+						},
+					},
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.APIServerCertAndKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  "tls.crt",
+									Path: path.Join("pki", kubeadmconstants.APIServerCertName),
+								},
+								{
+									Key:  "tls.key",
+									Path: path.Join("pki", kubeadmconstants.APIServerKeyName),
+								},
+							},
+						},
+					},
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.APIServerKubeletClientCertAndKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  "tls.crt",
+									Path: path.Join("pki", kubeadmconstants.APIServerKubeletClientCertName),
+								},
+								{
+									Key:  "tls.key",
+									Path: path.Join("pki", kubeadmconstants.APIServerKubeletClientKeyName),
+								},
+							},
+						},
+					},
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.ServiceAccountKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  "tls.crt",
+									Path: path.Join("pki", kubeadmconstants.ServiceAccountPublicKeyName),
+								},
+								{
+									Key:  "tls.key",
+									Path: path.Join("pki", kubeadmconstants.ServiceAccountPrivateKeyName),
+								},
+							},
+						},
+					},
+					{
+						Secret: &v1.SecretProjection{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: kubeadmconstants.FrontProxyCACertAndKeyBaseName,
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  "tls.crt",
+									Path: path.Join("pki", kubeadmconstants.FrontProxyCACertName),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func schedulerProjectedVolume() v1.Volume {
```

Comment:
could you put all this secret/secretvolume stuff in an other file to keep things clear?